.github/workflows/test.yaml

```diff
@@ -12,8 +12,18 @@ on:
     tags: ["**"]
   workflow_dispatch:
 
+defaults:
+  run:
+    # Both TigerVNC and TurboVNC reports "the input device is not a TTY" if
+    # started without a TTY. GitHub Actions environments doesn't come with one,
+    # so this provides one.
+    #
+    # ref: https://github.com/actions/runner/issues/241#issuecomment-842566950
+    #
+    shell: script --quiet --return --log-out /dev/null --command "bash -e {0}"
+
```

Comment:
@yuvipanda I think this may be what you need to get things working locally for you work also in github actions - it made the a difference to me.

I think also there are ipv4 / ipv6 differences, and I now know there were TigerVNC and TurboVNC differences with regards to listening on localhost or not by default, and that websockify will make things listen to localhost anyhow by intercepting the port bind system call or something like that - its messy!

Anyhow, with this, I got it working fully finally.